hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/read/TestECBlockReconstructedInputStream.java

```diff
@@ -0,0 +1,317 @@
+          long read = stream.read(buf, 0, buf.length);
+          totalRead += read;
+          Assert.assertEquals(expectedRead, read);
+          ECStreamTestUtil.assertBufferMatches(
+              ByteBuffer.wrap(buf, 0, (int)read), dataGenerator);
+        }
+        // Next read should be EOF
+        long read = stream.read(buf, 0, buf.length);
+        Assert.assertEquals(-1, read);
+      }
+    }
+  }
+
+  @Test
+  public void testSeek() throws IOException {
+    int readBufferSize = repConfig.getEcChunkSize() + 1024;
+    // 3 stripes and a partial chunk
+    int blockLength = repConfig.getEcChunkSize() * repConfig.getData() * 3
+        + repConfig.getEcChunkSize() - 1;
+    ByteBuffer[] dataBufs = allocateBuffers(repConfig.getData(),
+        repConfig.getEcChunkSize() * 4);
+    ECStreamTestUtil.randomFill(dataBufs, repConfig.getEcChunkSize(),
+        dataGenerator, blockLength);
+    ByteBuffer[] parity = generateParity(dataBufs, repConfig);
+    addDataStreamsToFactory(dataBufs, parity);
+
+    Map<DatanodeDetails, Integer> dnMap = createIndexMap(1, 2, 4, 5);
+    try(ECBlockReconstructedStripeInputStream stripeStream
+            = createStripeInputStream(dnMap, blockLength)) {
+      try (ECBlockReconstructedInputStream stream =
+               new ECBlockReconstructedInputStream(repConfig, stripeStream)) {
+        ByteBuffer b = ByteBuffer.allocate(readBufferSize);
+
+        int seekPosition = 0;
+        for (int i = 0; i < 100; i++) {
+          resetAndAdvanceDataGenerator(seekPosition);
+          long expectedRead = Math.min(stream.getRemaining(), readBufferSize);
+          long read = stream.read(b);
+          Assert.assertEquals(expectedRead, read);
+          ECStreamTestUtil.assertBufferMatches(b, dataGenerator);
+          seekPosition = random.nextInt(blockLength);
+          stream.seek(seekPosition);
+          b.clear();
+        }
+        // Seeking beyond EOF should give an error
+        try {
+          stream.seek(blockLength + 1);
+          Assert.fail("Seek beyond EOF should error");
+        } catch (IOException e) {
+          // expected
+        }
+      }
+    }
+  }
+
+  private void resetAndAdvanceDataGenerator(long position) {
+    dataGenerator = new SplittableRandom(randomSeed);
+    for (long i = 0; i < position; i++) {
+      dataGenerator.nextInt(255);
+    }
+  }
+
+  /**
+   * Returns a new map containing a random DatanodeDetails for each index in
+   * inputs.
+   * @param idxs A list of indexes to add to the map
+   * @return A map of DatanodeDetails to index.
+   */
+  private Map<DatanodeDetails, Integer> createIndexMap(int... idxs) {
+    Map<DatanodeDetails, Integer> map = new HashMap<>();
+    for (int i : idxs) {
+      map.put(MockDatanodeDetails.randomDatanodeDetails(), i);
+    }
+    return map;
+  }
+
+  /**
+   * Return a list of num ByteBuffers of the given size.
+   * @param num Number of buffers to create
+   * @param size The size of each buffer
+   * @return
+   */
```

Comment:
this can be a static method right. Probably we can move this to util class?

Reply:
Yea it can be, I have moved it into the util class.

Reply:
Thanks